catroid/src/main/java/org/catrobat/catroid/content/actions/WebRequestAction.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions;
+
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.badlogic.gdx.scenes.scene2d.Action;
+
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.InterpretationException;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.web.WebConnection;
+import org.catrobat.catroid.web.WebConnectionFactory;
+
+public class WebRequestAction extends Action implements WebConnection.WebRequestListener {
+
+	private static final float TIMEOUT_DURATION = 60f;
+	private static final Double ERROR_TIMEOUT = 504d;
+	private static final Double ERROR_TOO_MANY_REQUESTS = 429d;
+	private static final Double ERROR_BAD_REQUEST = 400d;
+
+	private Sprite sprite;
+	private Formula changeVariable;
+	private String url;
+	private UserVariable userVariable;
+	private WebConnectionFactory webConnectionFactory;
+
+	private boolean firstStart = true;
+	private float time = 0f;
+	private WebConnection webConnection = null;
+	private boolean isRequestFinished = false;
+	private String response = null;
+
+	public WebRequestAction() {
+	}
+
+	public void setUserVariable(UserVariable userVariable) {
+		if (userVariable == null) {
+			return;
+		}
+		this.userVariable = userVariable;
+	}
+
+	public void setUrl(String url) {
+		this.url = url;
+	}
+
+	public void setChangeVariable(Formula changeVariable) {
+		this.changeVariable = changeVariable;
+	}
+
+	public void setSprite(Sprite sprite) {
+		this.sprite = sprite;
+	}
+
+	public void setWebConnectionFactory(WebConnectionFactory webConnectionFactory) {
+		this.webConnectionFactory = webConnectionFactory;
+	}
+
+	public void interpretUrl() {
+		try {
+			this.url = changeVariable.interpretString(sprite);
+		} catch (InterpretationException interpretationException) {
+			Log.d(getClass().getSimpleName(), "Couldn't interpret formula", interpretationException);
+		}
+	}
+
+	@Override
+	public boolean act(float delta) {
+		if (firstStart) {
+			firstStart = false;
+			webConnection = webConnectionFactory.getWebConnection(url, this);
+			boolean successfullyAddedConnection = StageActivity.webConnectionHolder.addConnectionToList(webConnection);
+			if (!successfullyAddedConnection) {
+				userVariable.setValue(ERROR_TOO_MANY_REQUESTS);
+				return true;
+			}
+
+			boolean successfullySentWebRequest = webConnection.sendWebRequest();
```

Comment:
this boolean variables dont add any value for readability, for me they harm readability, because the fact that both start with "succefully..." doesnt help at all.
remove them and just have the webConnection.sendWebRequest() in the if clause.